services/checks/checks.go

```diff
@@ -131,13 +130,301 @@ func (s *Service) Run(ctx context.Context) {
 
 // Checks returns available checks.
 func (s *Service) Checks() []check.Check {
-	s.m.RLock()
-	defer s.m.RUnlock()
+	s.cm.Lock()
+	defer s.cm.Unlock()
 
 	r := make([]check.Check, 0, len(s.checks))
 	return append(r, s.checks...)
 }
 
+func (s *Service) processTasks(ctx context.Context, tasks []task) {
```

Comment:
Please start adding comments to private methods too. Once you start doing that, you may come up with better names and even ideas how to refactor it better.

In particular, I think this function should be simplified even more – just run query, wait for result (polling for result can be a separate function), execute script. Do not wait for several results in parallel, etc. Yes, it will be slow, but it will be simpler. Once we have a simple/linear/blocking implementation, we will make it more efficient.

Reply:
Comments added, the whole service simplified